app/models/setting/provisioning.rb

```diff
@@ -21,6 +21,7 @@ def self.load_defaults
         self.set('update_ip_from_built_request', N_("Foreman will update the host IP with the IP that made the built request"), false, N_('Update IP from built request')),
         self.set('use_shortname_for_vms', N_("Foreman will use the short hostname instead of the FQDN for creating new virtual machines"), false, N_('Use short name for VMs')),
         self.set('dns_conflict_timeout', N_("Timeout for DNS conflict validation (in seconds)"), 3, N_('DNS conflict timeout')),
+        self.set('clean_up_failed_deployment', N_("Foreman will delete virtual machine if provisioning script ends with non zero exit code"), false, N_('Clean up failed deployment')),
```

Comment:
I'd say this is useful enough behaviour to warrant the default being `true`, then people can disable it if they need to debug.